core/src/main/java/io/grpc/Contexts.java

```diff
@@ -130,4 +132,21 @@ public void onReady() {
       }
     }
   }
+
+  /**
+   * Returns the {@link Status} of a cancelled context or {@code null} if the context
+   * is not cancelled.
+   */
+  public static Status statusFromCancelled(Context context) {
```

Comment:
Consider moving this into Contexts so we can keep the separation I mentioned above (and support TimeoutException here)